### ISSUE TITLE
no-unused-vars in type checked JS file

### ISSUE THREAD

**Issue by OliverJAsh** [nzakas/eslint-plugin-typescript#28]
When using TypeScript’s type checking of JS files with the `checkJs` flag, ESLint will complain when an identifier is unused, when in fact it is used for type checking within a JSDoc:

``` js
import { StatelessComponent } from 'react'; // eslint error: 'StatelessComponent' is defined but never used. (no-unused-vars)

/** @type {StatelessComponent<any>} */ // StatelessComponent is used here
const MyComponent = () => {

};
```

I understand that this plugin will silence these errors for TypeScript files: https://github.com/nzakas/eslint-plugin-typescript/blob/420e9a96d612a49949f2a17c559e8a3ad0b00e7d/docs/rules/no-unused-vars.md

Can we do the same for JavaScript files?

**Comment by JamesHenry** [comment]
Sorry for the delay, @OliverJAsh. I think a feature like this would be a ways off.

I personally would advise that you use the relevant options in the TypeScript compiler to enforce "no unused" behaviour, rather than using ESLint for that purpose.

The TSC options are `noUnusedLocals` and `noUnusedParameters`. As this is a strong solution to the issue, I am going to close it for now.
